### ISSUE TITLE
Saving a project with lots of observed data takes ages.

### ISSUE THREAD

**Issue by sfrechen** [Open-Systems-Pharmacology/PK-Sim#1285]
Saving a project with lots of observed data takes ages. The progress bar stays at 40% for quite a while. Final project size only 25 MB.

**Comment by msevestre** [comment]
How many observed data are we talking about?
How much time are talking about?
Assuming this happens only the first time ? Upon reload of project, this should not be the case?

**Comment by sfrechen** [comment]
The file has around 2472 observed data objects.
Around 7.7 Megabyte in .json format.
Saving takes more than 10 minutes. Every time.

**Comment by msevestre** [comment]
2.5k observed data!! WOW
PK-Sim was definitely not designed with this number in mind. We do not track changes in obs data specifically so they are always saved again. Cost nothing for  < 100 observed data but this has clearly an impact on such large number

You will have other issues with PK-Sim down the road with so many data.
Is there a reason why you do not use a mean an std for those data?

**Comment by sfrechen** [comment]
> You will have other issues with PK-Sim down the road with so many data.
Unfortunatel yes. The file is pretty unstable.

> Is there a reason why you do not use a mean an std for those data?
Yes, I would like to use the individual data. That is what we work with. The package is just the complete dataset of all Phase I studies for a current development project.

Why does PK-Sim has soch much issues with it? 7 Megabytes of data is not a lot!

**Comment by msevestre** [comment]
@sfrechen This has nothing to do with the "amount" of data. We store many MB of simulated results. But the storage was optimized for this scenario

Each observed data is a specific entry in the database as opposed to a blob of data saved. This allows for quick read of specific dataset but clearly falls on its face when using this amount of data. 

I do not know how you plan to use this data within the tool, to be honest. PI, Plots etc.. will all probably barf at some point with this amount of data repositories

**Comment by StephanSchaller** [comment]
Why don't you use R or Matlab? ;-)
Ideal for this scenario!

**Comment by msevestre** [comment]
@sfrechen Can you send me the project? Or a project equivalent in terms of numbers of observed data? I could try to reproduce and at least see where we can optimize. Very hard to reproduce otherwise

**Comment by sfrechen** [comment]
> Why don't you use R or Matlab? ;-) (calling the PK-Sim Model)

We do. But we do also need a comprehensive file, e.g. for eSubmission.

Anyways, let's not put too much priority here. I guess we are planning to have in future data stored externally anyway, right? Within this implementation, we should optimize access to data then.

**Comment by msevestre** [comment]
I do not see how reading the data `every time` the project loads can be any faster! In your case, you may have to open hundreds of excel files. 

I'd like to get the example however

**Comment by sfrechen** [comment]
I would Need to prepare one with dummy data.

Well. Probably thousands of excel files is not the ideal way for any new implementation.

**Comment by msevestre** [comment]
>Well. Probably thousands of excel files is not the ideal way for any new implementation.

I agree. But so far, the specification was to always load the data from the original source (which is always an excel file). I do not know how you loaded your 2.5k in PK-Sim in the first place, but this loading by reference is supposed to `ONLY` work when using the normal importer flow.

This could be revisited obviously if the specifications have changed
@Yuri05 Please advise

**Comment by sfrechen** [comment]
I did import via conversion of NONMEM datafile via R to JSON (into 2.5k building blocks).

Yes, we need to revisit this. We need probably somewhat like filterings etc.
